tests/test-plugins/activity-container/src/main.rs

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Plugin for querying how long it has been since a commit was last made to a repo
+
+use clap::Parser;
+use hipcheck_sdk::{prelude::*, types::Target};
+use jiff::Timestamp;
+use serde::Deserialize;
+use std::{result::Result as StdResult, sync::OnceLock};
+
+#[derive(Deserialize)]
+struct Config {
+	weeks: Option<u16>,
+}
+
+static CONFIG: OnceLock<Config> = OnceLock::new();
+
+/// Returns the span of time since the most recent commit to a Git repo as `jiff:Span` displayed as a String
+/// (Which means that anything expecting a `Span` must parse the output of this query appropriately)
+#[query(default)]
+async fn activity(engine: &mut PluginEngine, target: Target) -> Result<String> {
+	log::debug!("running activity query");
+
+	let repo = target.local;
+
+	// Get today's date
+	let today = Timestamp::now();
+
+	// Get the date of the most recent commit.
+	let value = engine
+		.query("mitre/git/last_commit_date", repo)
+		.await
+		.map_err(|e| {
+			log::error!("failed to get last commit date for activity metric: {}", e);
+			Error::UnspecifiedQueryState
+		})?;
+
+	let Value::String(date_string) = value else {
+		return Err(Error::UnexpectedPluginQueryInputFormat);
+	};
+	let last_commit_date: Timestamp = date_string.parse().map_err(|e| {
+		log::error!("{}", e);
+		Error::UnspecifiedQueryState
+	})?;
+
+	// Get the time between the most recent commit and today.
+	let time_since_last_commit = today.since(last_commit_date).map_err(|e| {
+		log::error!("{}", e);
+		Error::UnspecifiedQueryState
+	})?;
+
+	Ok(time_since_last_commit.to_string())
+}
+
+#[derive(Clone, Debug)]
+struct ActivityPlugin;
+
+impl Plugin for ActivityPlugin {
+	const PUBLISHER: &'static str = "mitre";
+
+	const NAME: &'static str = "activity";
```

Comment:
Rename to `activity-container` just to differentiate from the non-containerized one if, for some reason, both were running and we needed to debug issues.